mongo/client.go

```diff
@@ -1102,6 +1102,11 @@ func (c *Client) NumberSessionsInProgress() int {
 	return c.sessionPool.CheckedOut()
 }
 
+// Timeout returns the timeout set for this client.
+func (c *Client) Timeout() *time.Duration {
+	return c.timeout
+}
```

Comment:
This is the first time we're providing access to a `Client` config value. Does this pattern work for potentially providing access to other `Client` config values?

Reply:
It's a good question. I think it's an odd pattern, but accessing the `timeout` value from the `gridfs` package is not (?) possible without it. My only concern is that users will make use of the public getter for their own logic.

Reply:
I suspect users will start to use the `Timeout` function to get access to the client timeout value. I don't see an obvious way to get around that requirement other than to define a new function on `*mongo.Database` like `client.Database("name").GridFSBucket()` and document that client-level timeouts only apply to GridFS buckets if using that function. That seems very confusing, so I don't recommend it (for the current Go Driver version, at least).

I think the existing `Timeout` function is probably the best path forward.